commands/commands.js

```diff
@@ -3,31 +3,107 @@ import promptSync from "prompt-sync";
 import { constantValues } from "../constants/consts.js";
 import { shortcuts } from "../constants/shortcut.js";
 import { listCategories } from "../api/category.js";
+import { getJoke } from "../api/joke.js";
+import * as fs from "fs";
+import { finalData } from "../data/finalData.js";
+import { fileURLToPath } from 'url';
+import { dirname } from 'path';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+
 
 export const prompt = promptSync();
 
-const showWelcomeBanner = () => {
-    console.log("\n \n " + chalk.green(constantValues.title) + "\n \n" + chalk.bgGreen(constantValues.description) + "\n\n" + constantValues.jokeCategories + chalk.red(shortcuts.see_joke_categories) + "\n" + constantValues.backToStart + chalk.red(shortcuts.go_back_to_start) + "\n" + constantValues.exitProgram + chalk.red(shortcuts.exit_project) + "\n" + constantValues.viewJokesDatabase + chalk.red(shortcuts.view_jokes_database) + "\n");
+export const showAppTitle = () => {
+    console.log("\n \n " + chalk.green(constantValues.title));
+}
+
+export const showWelcomeBanner = () => {
+    console.log("\n \n" + chalk.bgGreen(constantValues.description) + "\n\n" + constantValues.jokeCategories + chalk.red(shortcuts.see_joke_categories) + "\n" + constantValues.exitProgram + chalk.red(shortcuts.exit_project) + "\n" + constantValues.viewJokesDatabase + chalk.red(shortcuts.view_jokes_database) + "\n" + constantValues.askForHelp + chalk.red(shortcuts.ask_for_help) + "\n");
 }
 
 export const askName = () => {
-    showWelcomeBanner();
     console.log(chalk.green("What is your name?"));
     const name = prompt(">>");
+    finalData.name = name;
     return name;
 }
 
+const storeData = () => {
+    const userDetails = JSON.stringify(finalData);
+    const readerFile = fs.existsSync(`${__dirname}/../files/${finalData.name}.json`, "utf-8");
+
+    if (!readerFile) {
+        // console.log("doesntExist");
+        // console.log(readerFile);
+        fs.writeFileSync(`${__dirname}/../files/${finalData.name}.json`, userDetails, (err) => {
+            if (err) {
+                console.log("error")
+            }
+            console.log("\n" + "Your jokes are saved on " + chalk.blue(finalData.name) + chalk.blue(".json") + ". Please enter " + chalk.blue("view") + " to view the file.");
+
+        });
+        console.log("\n" + "Your jokes are saved on " + chalk.blue(finalData.name) + chalk.blue(".json") + ". Please enter " + chalk.blue("view") + " to view the file.");
+    }
+    else {
+        // console.log("exists");
+        // console.log(userDetails);
+        fs.appendFileSync(`${__dirname}/../files/${finalData.name}.json`, userDetails, (err) => {
```

Comment:
Maybe here we could do something like this by adding some line breaks that helps in reading the files later . 
```suggestion
        fs.appendFileSync(`${__dirname}/../files/${finalData.name}.json`, `\n${userDetails}`, (err) => {
```

Reply:
Why are we giving a line break before the userDetails and not afterwards?
When we append the file, the newer data gets stores after whatever is already available, no?

Reply:
This is the case for append , so when appending giving space before adds a space initially making the added line move a line below .